src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java

```diff
@@ -26,13 +32,22 @@ public class AnnotationConfigApplicationContext implements ApplicationContext {
     public AnnotationConfigApplicationContext(String packageName) {
         Objects.requireNonNull(packageName);
         this.packagePath = packageName;
-        scan();
+        initiateContext();
     }
 
     /**
-     * Scanning a package for classes or methods annotated by @Component and @Bean
+     * Scanning a package for classes annotated by {@link Component}.
+     * It creates an instances of such classes and put them into context.
      */
-    private void scan() {
+    @SneakyThrows
```

Comment:
I think we should add a proper error handling instead of using `@SneakyThrows`.

Reply:
Agree, fixed